### ISSUE TITLE
Add "%ne" format support for forge console log

### ISSUE THREAD

**Issue by red-swan** [foundry-rs/foundry#4841]
### Component

Forge

### Describe the feature you would like

Tokens in Solidity having varying decimals (18 for many ERC20, 8 for wBTC, etc.). This makes it difficult to recognize exactly how much of something is being printed when shown in "%d" or "%e" format. I'd like to have "%ne" supported to see these numbers as decimal.

E.g.
Someone's supply of wBTC on chain: 420000000
Formatted using "%8e": 4.2

### Additional context

There was [an idea](https://github.com/foundry-rs/foundry/issues/4464#issuecomment-1455114406) to add this as part of #4464, but was not picked up. I'm reiterating the request here.

**Comment by RickVM** [comment]
I'd like to have this feature as well and am considering an implementation.
Would a PR from my side be appreciated? Must admint i'm a total rust newbie but it seems like it shouldn't be too hard based on the %e PR

**Comment by mattsse** [comment]
I see so exponential should take an optional number argument

we could support this by using an optional number value here:

https://github.com/foundry-rs/foundry/blob/5494c33bc7977b3537bd296e375431d938d44ca3/crates/common/src/fmt/console.rs#L16-L17

for parsing we need to make some changes to support lookahead

https://github.com/foundry-rs/foundry/blob/5494c33bc7977b3537bd296e375431d938d44ca3/crates/common/src/fmt/console.rs#L245-L246

we either need to use `.peekable` for the iterator or track the case when a number follows a `%` manually

https://github.com/foundry-rs/foundry/blob/5494c33bc7977b3537bd296e375431d938d44ca3/crates/common/src/fmt/console.rs#L237-L237

basically we need to parse

`<number>e` and convert this into `ExponentialNumber(number)`

**Comment by zerosnacks** [comment]
Hi @RickVM, would be great to get this in. Let me know if you are running into any blockers - we are happy to give pointers / help out

**Comment by zerosnacks** [comment]
No problem! Good luck on your project launch! I'll pick up the ticket

**Comment by qiweiii** [comment]
Hi, I would like to try impl this
